### PR TITLE
[move] improve better error message for const bindings

### PR DESCRIPTION
**Pull request by cgswords**
## Description 

This adds a slightly more-informative error message to constant parsing failures based on feedback from @tedks 

## Test Plan 

Added a new test

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: external-crates/move/move-compiler/src/diagnostics/codes.rs [hunk 1]
```diff
@@ -182,6 +182,8 @@ codes!(
         InvalidSpecBlockMember: { msg: "invalid spec block member", severity: NonblockingError },
         InvalidRestrictedIdentifier:
             { msg: "invalid identifier escape", severity: NonblockingError },
+        NeedsTypeAnnotation:
+            { msg: "Requires a type annotation 'name: type ='", severity: BlockingError },
```
Comment (tnowacki) [inline external-crates/move/move-compiler/src/diagnostics/codes.rs:186]: I think we should probably just keep this as `UnexpectedToken` but just with a better error message. The codes are broad categories generally, and not normally this specific.

File: external-crates/move/move-compiler/src/parser/syntax.rs [hunk 2]
```diff
@@ -2435,7 +2435,12 @@ fn parse_constant_decl(
     }
     consume_token(context.tokens, Tok::Const)?;
     let name = ConstantName(parse_identifier(context)?);
-    consume_token(context.tokens, Tok::Colon)?;
+    if consume_token(context.tokens, Tok::Colon).is_err() {
```
Comment (tnowacki) [inline external-crates/move/move-compiler/src/parser/syntax.rs:2438]:
```suggestion
    if context.tokens.peek() != Tok::Colon {
```

Comment (tnowacki) [reply external-crates/move/move-compiler/src/parser/syntax.rs:2438]: consume_token is just a helper to generate an error, no need to do that just to through it away

Comment (tnowacki) [inline external-crates/move/move-compiler/src/parser/syntax.rs:2438]: Alternatively, we might want to make a helper `consume_token!(context.tokens, Tok::Colon, "Add type annotation to this constant")`
or `consume_token!(context.tokens, Tok::Colon, "Add type annotation to this constant")`

`consume_token_with(context.tokens, Tok::Colon, || "Expected a type annotation for this constant")`

File: external-crates/move/move-compiler/tests/move_check/parser/constant_type_annotation_invalid.exp [hunk 3]
```diff
@@ -0,0 +1,6 @@
+error[E01013]: Requires a type annotation 'name: type ='
+  ┌─ tests/move_check/parser/constant_type_annotation_invalid.move:2:11
+  │
+2 │     const EThreadMaxxed = 1;
+  │           ^^^^^^^^^^^^^ Add type annotation to this constant
```
Comment (tnowacki) [inline external-crates/move/move-compiler/tests/move_check/parser/constant_type_annotation_invalid.exp:5]: I would probably switch this to pointing to the `=` and saying `"Expected a type annotation for this constant, e.g. 'EThreadMaxxed: <type>'"` or something

**Review by tnowacki — COMMENTED** [review]
Absolutely love the PR! 

Just lots of little thoughts about generalizing this and making it feel more inline with other errors, but while also making it more helpful :)

**Review by tnowacki — COMMENTED** [review]
https://github.com/MystenLabs/sui/blob/main/external-crates/move/move-compiler/src/shared/mod.rs#L158
We should add a suppression for this.

**Review by tzakian — APPROVED** [review]
Love the macro! I can't tell you hoy many times I wish I had something like this when working in the parser...

**Pull request merged**